### ISSUE TITLE
Lint/Debugger complains about using "console" when it is not appropriate

### ISSUE THREAD

**Issue by drbrain** [rubocop-hq/rubocop#9500]
If I define and use a `console` method, `attr_reader :console` or a minitest/spec `let(:console)` Lint/Debugger says I should "Remove debugger entry point console".

I have not `require`d a debugger, nor have I added one to a `Gemfile`, so I cannot be entering a debugger, I'm working with a method or accessor that is appropriately named for my usage and is in-scope.  No `method_missing` magic is involved.

## Expected behavior

Lint/Debugger does not complain by default about methods I define that happen collide with a debugging tool I have not even loaded a gem for.

I expect zero lint warnings for the code included below.

## Actual behavior

```
$ cat t.rb
require "stringio"
require "minitest/spec"

describe "bug" do
  let(:console) { StringIO.new }
  let(:thing) { [console] }

  it "does not complain about console" do
    assert thing
  end
end

class T1
  def console
  end

  def test
    console.puts "this is not a debugger"
  end
end

class T2
  attr_reader :console

  def test
    console.puts "this is not a debugger"
  end
end

module T3
  def console
    $stdeout
  end
end

class T4
  include T3

  def test
    console.puts "this is not a debugger"
  end
end

$ rubocop -l t.rb
Inspecting 1 file
W

Offenses:

t.rb:6:18: W: Lint/Debugger: Remove debugger entry point console.
  let(:thing) { [console] }
                 ^^^^^^^
t.rb:18:5: W: Lint/Debugger: Remove debugger entry point console.
    console.puts "this is not a debugger"
    ^^^^^^^
t.rb:26:5: W: Lint/Debugger: Remove debugger entry point console.
    console.puts "this is not a debugger"
    ^^^^^^^
t.rb:40:5: W: Lint/Debugger: Remove debugger entry point console.
    console.puts "this is not a debugger"
    ^^^^^^^

1 file inspected, 4 offenses detected
$ ls -a
.	..	t.rb
```

## Steps to reproduce the problem

Run `rubocop -l` on test embedded above

## RuboCop version

```
$ rubocop -V
1.9.1 (using Parser 3.0.0.0, rubocop-ast 1.3.0, running on ruby 2.6.6 x86_64-darwin19)
```

**Comment by dvandersluis** [comment]
RuboCop is not currently aware of where a method being called is defined, so your various ways of defining `console` are clearly false positives, but there isn't an easy way of resolving that.

That being said, when #8929 changed how `Lint/Debugger` is configured, it went from registering `binding.console` (for example) to `console` with any receiver, which is clearly wrong here, so I think we can fix it to use an explicit receiver again. @bbatsov thoughts?

**Comment by dvandersluis** [comment]
Actually looks like #7636 fixed this a while ago (for `console` specifically) and then it too was broken by #8929.
